src/main/java/org/gridsuite/modification/server/service/NetworkModificationService.java

```diff
@@ -2242,4 +2308,62 @@ public Mono<Void> updateLineAttachToVoltageLevel(UUID modificationUuid, LineAtta
         return Mono.empty();
     }
 
+    private Identifiable<?> getEquipmentByIdentifiableType(Network network, String type, String equipmentId) {
+        if (type == null || equipmentId == null) {
+            return null;
+        }
+        switch (IdentifiableType.valueOf(type)) {
+            case HVDC_LINE:
+                return network.getHvdcLine(equipmentId);
+            case LINE:
+                return network.getLine(equipmentId);
+            case TWO_WINDINGS_TRANSFORMER:
+                return network.getTwoWindingsTransformer(equipmentId);
+            case THREE_WINDINGS_TRANSFORMER:
+                return network.getThreeWindingsTransformer(equipmentId);
+            case GENERATOR:
+                return network.getGenerator(equipmentId);
+            case LOAD:
+                return network.getLoad(equipmentId);
+            case BATTERY:
+                return network.getBattery(equipmentId);
+            case SHUNT_COMPENSATOR:
+                return network.getShuntCompensator(equipmentId);
+            case STATIC_VAR_COMPENSATOR:
+                return network.getStaticVarCompensator(equipmentId);
+            case DANGLING_LINE:
+                return network.getDanglingLine(equipmentId);
+            case HVDC_CONVERTER_STATION:
+                return network.getHvdcConverterStation(equipmentId);
+            case SUBSTATION:
+                return network.getSubstation(equipmentId);
+            case VOLTAGE_LEVEL:
+                return network.getVoltageLevel(equipmentId);
+            case BUSBAR_SECTION:
+                return network.getBusbarSection(equipmentId);
+            default:
+                return null;
+        }
+    }
+
+    private Terminal getTerminalFromIdentifiable(Network network,
+                                                 String equipmentId,
+                                                 String type,
+                                                 String voltageLevelId) {
+        if (network != null && equipmentId != null && type != null && voltageLevelId != null) {
+            Identifiable<?> identifiable = getEquipmentByIdentifiableType(network, type, equipmentId);
+
+            if (identifiable == null) {
+                throw new NetworkModificationException(EQUIPMENT_NOT_FOUND);
+            }
+
+            if (identifiable instanceof Injection<?>) {
+                return ((Injection) identifiable).getTerminal();
```

Comment:
IntelliJ advises to "specialize" / line-up the cast to Injection<?>. The same for `Branch<?>` below.